src/classes/actions-cache/retry.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  HttpClientError,
+  HttpClientResponse,
+  HttpCodes
+} from '@actions/http-client';
+import {
+  isServerErrorStatusCode,
+  TypedResponseWithError
+} from './http-responses';
+import * as core from '@actions/core';
+
+const isRetryableStatusCode = (statusCode?: number): boolean => {
+  if (!statusCode) {
+    return false;
+  }
+  const retryableStatusCodes = [
+    HttpCodes.BadGateway,
+    HttpCodes.ServiceUnavailable,
+    HttpCodes.GatewayTimeout
+  ];
+  return retryableStatusCodes.includes(statusCode);
+};
+
+const sleep = (milliseconds: number): Promise<void> =>
+  new Promise(resolve => setTimeout(resolve, milliseconds));
+// The default number of retry attempts.
+const DefaultRetryAttempts = 2;
+// The default delay in milliseconds between retry attempts.
+const DefaultRetryDelay = 5000;
+
+const retry = async <T>(
+  name: string,
```

Comment:
Let's represent parameters with an object when there're so many of them. There's no way one would remember this order. Same goes for other functions in this file.

Reply:
It is inherited code as well. It has its own code-styling and motivations. I prefer to keep it as is until the refactoring.